notes/DATABASE.md

```diff
@@ -1,3 +1,51 @@
 # Database Systems and Data Thinking
 
-_Notes for this topic currently under development_
\ No newline at end of file
+## Questions to Consider When Thinking About Data
+* What is my data?
+* How is my data structured?
+* How I can relate different parts of data?
+* How much data will I need to store?
+* How frequently will new data be added, existing data be changed and read?
+* How frequently and how drastically will the structure of my data change as my product grows?
+
+## Case Studies
+* TODO: some example where relational DB is clear favorite
+* TODO: some example where NoSQL DB is clear favorite
+
+## Relational Databases
+* Pros
+** Structured
+** ACID-safe
+** Tables can be related to one another
+** Available DBs are fairly consistent and standardized, strengths and weaknesses are somewhat uniform
+* Cons
+** Structured nature can cause additional overhead
+** Need to consider normalization forms
+* Some Examples - And When to Use Them
+** Oracle - I can pay for the licensing fee! And I want something generally stable and performant
+** MySQL/Maria - I can't pay for the licensing fee! And I want something generally stable and performant
+** PostgreSQL - I need to lots of data sharding and store large formats (blobs, big files)
+** SQLite - I need something simple that I can set up quickly
+** Microsoft SQL - I am using a Microsoft tech stack
+
+## NoSQL Databases
+* Pros
+** Fast AF, especially when used in the right situation
+** Able to handle unstructured data
+* Cons
+** Need to be able to handle unstructured data
+** Many aren't ACID-safe by default, or require performance hits to do so
```

Comment:
found a cool chart that you might be able to reference http://blog.scottlogic.com/2014/08/04/mongodb-vs-couchdb.html